compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -119,70 +95,49 @@ extractDocs extractOpts diagsLogger ideOpts fp = do
                 = MS.elems . MS.withoutKeys typeMap . Set.unions
                 $ dc_templates : MS.elems dc_choices
 
-            md_adts = mapMaybe (filterTypeByExports ctx) adts
+            md_adts = mapMaybe (filterTypeByExports dc_exports) adts
 
         in ModuleDoc {..}
 
 -- | This is equivalent to Haddock’s Haddock.Interface.Create.collectDocs
-collectDocs :: [LHsDecl a] -> [(LHsDecl a, Maybe DocText)]
-collectDocs = go Nothing []
+collectDocs :: [LHsDecl GhcPs] -> [DeclData]
+collectDocs ds
+    | (nextDocs, decl:ds') <- spanMaybe getNextOrPrevDoc ds
```

Comment:
This looks like we will attach a `DocCommentPrev` at the beginning to the next declaration which seems odd. I _think_ this is the same behavior as before but it seems weird. Is there a case where this actually makes sense?

Reply:
I'm not sure if it comes up in practice, but if a `DocCommentNext` is followed up by a `DocCommentPrev` ... or we have a `DocCommentPrev` at the beginning of the list ... this will treat the `DocCommentPrev` as a `DocCommentNext` instead of silently ignoring it or treating it as a standalone declaration.